File: backend_api/CalendarShop.Api/Models/Favorite.cs
namespace CalendarShop.Api.Models;

public class Favorite
{
    public int FavoriteId { get; set; }
    public int UserId { get; set; }
    public int ProductId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User? User { get; set; }
    public Product? Product { get; set; }
}

File: mobile_flutter/lib/features/favorite/domain/entities/favorite.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'favorite.freezed.dart';

@freezed
class Favorite with _$Favorite {
  const factory Favorite({
    required int favoriteId,
    required int productId,
    required String productName,
    String? imageUrl,
    required double price,
    required String calendarType,
    required String productStatus,
    required DateTime createdAt,
  }) = _Favorite;
}
